tools/cgdiff/include/DiffFormatter.h

```diff
@@ -0,0 +1,155 @@
+/**
+ * File: CGDiff.cpp
+ * License: Part of the MetaCG project. Licensed under BSD 3 clause license. See LICENSE.txt file at
+ * https://github.com/tudasc/metacg/LICENSE.txt
+ */
+
+#pragma once
+
+#include "GlobalMDDiff.h"
+#include "NodeDiff.h"
+
+struct DiffFormatter {
+  /**
+   * Emit json representation of a list of NodeDiffs.
+   *
+   * @param nodeDiffs Vector of NodeDiff objects to format
+   * @param ignoring  List of difference categories that were ignored
+   * @return Formatted string summarizing the differences
+   */
+  static nlohmann::ordered_json emitAsJson(const std::vector<std::unique_ptr<Diff>>& diffs,
+
+                                           const std::vector<std::string>& ignoring, std::string_view cgA,
+                                           std::string_view cgB) {
+    nlohmann::ordered_json root;
+    root["cgA"] = cgA;
+    root["cgB"] = cgB;
+    root["ignoring"] = ignoring;
+
+    // collect node diffs and global metadata diffs separately
+    nlohmann::ordered_json nodeDiffs = nlohmann::ordered_json::object();
+    nlohmann::ordered_json globalMetaDiffs = nlohmann::ordered_json::object();
+
+    for (const auto& diff : diffs) {
+      nlohmann::ordered_json inner = diff->toJson();  // each Diff subclass implements its own toJson()
```

Comment:
```suggestion
      // each Diff subclass implements its own toJson()
      nlohmann::ordered_json inner = diff->toJson();
```
I personally prefer this way to attach comments. Gives the formatter a better time.